pkg/sql/catalog/descpb/structured.proto

```diff
@@ -534,6 +541,16 @@ message ConstraintToUpdate {
   reserved 5;
   optional uint32 not_null_column = 6 [(gogoproto.nullable) = false, (gogoproto.casttype) = "ColumnID"];
   optional UniqueWithoutIndexConstraint unique_without_index_constraint = 7 [(gogoproto.nullable) = false];
+  // `index` is only populated for `PRIMARY_KEY` or `UNIQUE` constraint.
+  optional IndexDescriptor index = 8 [(gogoproto.nullable) = false];
+  // `index_backed_constraint_validity` is the validity of this
+  //  index-backed constraint (PRIMARY KEY or UNIQUE).
+  // It will be
+  // - VALIDATED when the index is public;
+  // - VALIDATING when the index is being added (i.e. live on mutation slice with ADD direction);
+  // - DROPPING when the index is being dropped (i.e. live on mutation slice with DROP direction);
+  // - UNVALIDATED in no case!
+  optional ConstraintValidity index_backed_constraint_validity = 9 [(gogoproto.nullable) = false];
```

Comment:
I will change as your suggested! However, I think we cannot get rid of `ConstraintToUpdateDesc()` because this is needed if we want to modify the underlying the protobuf. Now with `index` implementing `Constraint`, we will need to add another method, maybe called `IndexDesc() *descpb.IndexDescriptor`, if the to-be-modified constraint is backed by an index.

One such scanrio of needing to modify the underlying protobuf is setting the name of a constraint. This will show up in both RENAME CONSTRAINT or, more generally, the adding/dropping path of a ConstraintName element.